x-pack/plugins/security_solution/server/lib/telemetry/endpoint_task.ts

```diff
@@ -193,28 +248,65 @@ export class TelemetryEndpointTask {
       const endpointAgentId = endpoint.endpoint_agent;
 
       const policyInformation = fleetAgents.get(fleetAgentId);
-      if (policyInformation?.policy_id) {
-        policyConfig = endpointPolicyCache.get(policyInformation?.policy_id);
+      if (policyInformation) {
+        policyConfig = endpointPolicyCache.get(policyInformation);
+
         if (policyConfig) {
           failedPolicy = policyResponses.get(policyConfig?.id);
         }
       }
 
       return {
+        '@timestamp': executeTo,
         agent_id: fleetAgentId,
         endpoint_id: endpointAgentId,
+        endpoint_version: endpoint.endpoint_version,
+        endpoint_package_version: policyConfig?.package?.version || null,
         endpoint_metrics: {
+          cpu: {
+            histogram: endpoint.endpoint_metrics.Endpoint.metrics.cpu.endpoint.histogram,
+            latest: endpoint.endpoint_metrics.Endpoint.metrics.cpu.endpoint.latest,
+            mean: endpoint.endpoint_metrics.Endpoint.metrics.cpu.endpoint.mean,
+          },
+          memory: {
+            latest: endpoint.endpoint_metrics.Endpoint.metrics.memory.endpoint.private.latest,
+            mean: endpoint.endpoint_metrics.Endpoint.metrics.memory.endpoint.private.mean,
+          },
+          uptime: {
+            endpoint: endpoint.endpoint_metrics.Endpoint.metrics.uptime.endpoint,
+            system: endpoint.endpoint_metrics.Endpoint.metrics.uptime.system,
+          },
+        },
+        endpoint_meta: {
           os: endpoint.endpoint_metrics.host.os,
-          cpu: endpoint.endpoint_metrics.Endpoint.metrics.cpu,
-          memory: endpoint.endpoint_metrics.Endpoint.metrics.memory,
-          uptime: endpoint.endpoint_metrics.Endpoint.metrics.uptime,
         },
-        policy_config: policyConfig,
-        policy_failure: failedPolicy,
+        policy_config: policyConfig !== null ? policyConfig?.inputs[0].config.policy : {},
+        policy_response:
+          failedPolicy !== null && failedPolicy !== undefined
+            ? {
+                agent_policy_status: failedPolicy?._source.event.agent_id_status,
+                manifest_version:
+                  failedPolicy?._source.Endpoint.policy.applied.artifacts.global.version,
+                status: failedPolicy?._source.Endpoint.policy.applied.status,
+                actions: failedPolicy?._source.Endpoint.policy.applied.actions
+                  .map((action) => (action.status !== 'success' ? action : null))
+                  .filter((action) => action !== null),
+              }
+            : {},
+        telemetry_meta: {
+          metrics_timestamp: endpoint.endpoint_metrics['@timestamp'],
```

Comment:
Nice this is great to know. It'll help us keep track if something is wonky w/ how the endpoint is publishing metrics.